docs/games.md

```diff
@@ -63,6 +64,17 @@ Status                                       | Game
 *   2 players.
 *   [Wikipedia](https://en.wikipedia.org/wiki/Backgammon)
 
+### Battleship
```

Comment:
Do you happen to have the game size anywhere? In particular, the number of histories & the number of infostates (basically everything a reviewer would ask for).

Reply:
I know the number of nodes/infosets/sequences for the small game, but not for the default game on the 10x10 board. It seems like the number of nodes would be a very large number though, because even just from the shots, there are (100 cells)^(100 shots) = 10^200 possibilities. And that does not take into account the positioning of the ships, which is something in the order of (2 orientation * 100 cells)^(10 ships) = 2 * 10^20. So I have a rough estimate of 10^220 states.

Reply:
I think that'd be great if you could add this, even with the caveats- I don't think the exact number matters, as it's still going to be a huge game. Just something so we can cite this in a review and say that the author of the game made the estimate (rather than me trying to do so).

Reply:
Sounds good! Would you add this information in `games.md`, or in `battleship.h`?
It looks like usually the size information is not put into the game list.

Reply:
`battleship.h` is a much better location- I should have made this comment there!

Reply:
I've added a whole section on the sizes of a few small games. Unfortunately those sizes come from my other generator, because I didn't know how to compute the number of sequences / information sets / terminal states idiomatically in OpenSpiel.

So, now I'd like to add tests that check that this PR's generator produces games with the same dimensions. Does OpenSpiel implement functions to compute the number of information sets / sequences / terminal states or will I need to implement that?

+@lanctot

Reply:
There is this code: 

https://source.corp.google.com/piper///depot/google3/third_party/open_spiel/examples/count_all_states.cc?q=count_all_states%20open_spiel

However, `GetAllStates` does a BFS over the entire game, so won't work for large games.

That grid looks awesome, thanks for doing that.

Reply:
> There is this code:
> 
> https://source.corp.google.com/piper///depot/google3/third_party/open_spiel/examples/count_all_states.cc?q=count_all_states%20open_spiel

Ahah Google did not let me go very far on `source.corp.google.com` :eyes:

I found the file you mentioned though, I will quickly add some tests by tonight :-)

Reply:
Ah, geez. Sorry for that. Any chance you can edit that out from your comment? I (obviously) shouldn't have posted that.

Glad you found it though!

Reply:
Oh I see, of course. I removed the comment with the quote.

Reply:
@finbarrtimbers maybe we should track the values for games somewhere in the repos, just as a reference? ... as one large comment in count_all_states? :) Or as an open document spreadsheet or public one?

Reply:
Either works for me, do you have a preference? I have a spreadsheet that I've shared internally, happy to open it up. 

We could also add it to the Arxiv paper.

Reply:
Thanks for removing that, @gabrfarina!

Reply:
@finbarrtimbers I have a question about the example code you pointed me to:

why does this line check strict inequality? Was that intended?

https://github.com/deepmind/open_spiel/blob/de2c9ef57038829c2709becb0104e3b2d76725cc/open_spiel/examples/count_all_states.cc#L43

Reply:
That's a bug. Can you fix that to `>= 0`?

Reply:
Sure, I don't mind :-) However, I'm not sure that moving to `>= 0` fixes the problem. If I'm not mistaken, the OpenSpiel API does not guarantee that infostates for different players return different strings, so before putting strings in the `hash_set`, I will prepend the player id to disambiguate. Does that sound reasonable?

+@lanctot If it all sounds good, I will go ahead an open another PR to address changes to `count_all_states.cc`

Reply:
I submitted a fix internally as that's probably quicker than opening a separate PR. It's already been merged internally (thanks @lanctot).

That is guaranteed in `api_test.py`:

https://github.com/deepmind/open_spiel/blob/de2c9ef57038829c2709becb0104e3b2d76725cc/open_spiel/integration_tests/api_test.py#L232

Reply:
That's fine by me. It is not enforced but it is true in practice. And we should add it to the docs if we haven't already.  We might have a test for it, but if not we should add one, at least for the games in the repos (allowing people to do otherwise in general) because several algorithms assume the info states are unique across players and use them as a key for storing in a table (i.e. CFR).

Reply:
I see, good to know. Perhaps it's worth remarking in `spiel.h` as well? I believe  that the condition holds also for Battleship, though it's easy to overlook: at the time of placing their own ship, no player has observed anything, so it seems tempting to use the same "empty" infostate string for both, and in the early commits that was in fact the case for this Generator.

Reply:
Ok this is excellent feedback, thanks! I've added it to my TODO.